centreon/www/class/centreon-clapi/centreonLDAP.class.php

```diff
@@ -264,65 +242,69 @@ public function addserver($parameters)
         }
 
         $serverList = $this->getLdapServers($arId);
-        $newServer = array('host_address' => $address, 'host_port' => $port);
-        if (in_array($newServer, $serverList)) {
+        $newServer = ['host_address' => $address, 'host_port' => $port];
+        if (in_array($newServer, $serverList, true)) {
             throw new CentreonClapiException(self::OBJECTALREADYEXISTS . ' ' . $address);
         }
 
         $this->db->query(
-            "INSERT INTO auth_ressource_host (auth_ressource_id, host_address, host_port, use_ssl, use_tls)
-             VALUES (:arId, :address, :port, :ssl, :tls)",
-            array(
+            'INSERT INTO auth_ressource_host (auth_ressource_id, host_address, host_port, use_ssl, use_tls)
+             VALUES (:arId, :address, :port, :ssl, :tls)',
+            [
                 ':arId' => $arId,
                 ':address' => $address,
                 ':port' => $port,
                 ':ssl' => $ssl,
-                ':tls' => $tls
-            )
+                ':tls' => $tls,
+            ]
         );
     }
 
     /**
-     * Delete configuration
+     * Delete configuration.
      *
      * @param int $parameters
+     * @param null|mixed $arName
+     *
      * @throws CentreonClapiException
      */
-    public function del($arName = null)
+    public function del($arName = null): void
     {
-        if (!isset($arName)) {
+        if (! isset($arName)) {
             throw new CentreonClapiException(self::MISSINGPARAMETER);
         }
         $arId = $this->getLdapId($arName);
         if (is_null($arId)) {
             throw new CentreonClapiException(self::OBJECT_NOT_FOUND . ' ' . $arName);
         }
-        $this->db->query("DELETE FROM auth_ressource WHERE ar_id = ?", array($arId));
+        $this->db->query('DELETE FROM auth_ressource WHERE ar_id = ?', [$arId]);
```

Comment:
Please use a prepare + execute to avoid this legacy call to a modified query signature from PDO 🙏